### PR TITLE
Adding Events Delegates Filter via Subpanel

### PR DESCRIPTION
**Pull request by shogunpol**
## Description

This functionality is in development and this PR will be closed when the contributor has updated their code.  

## Motivation and Context

Provides the ability to search/filter event delegates from within the delegate subpanel

## How To Test This

Navigate to an Event's DetailView and add delegates (if necessary).
Click on the Select Delegates dropdown action menu and click 'Filter'.
This will present with search fields of First Name and Last Name.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -42,6 +42,10 @@
  * display the words  "Powered by SugarCRM" and "Supercharged by SuiteCRM".
  */
 
+//error_reporting(E_ALL);
```
Comment (mattlorimer) [inline index.php]: index.php should not be modified with comment out code

File: include/SubPanel/SubPanel.js [hunk 2]
```diff
@@ -62,7 +62,7 @@ function submitSearch(subpanel) {
     });
 
     var url = 'index.php?sugar_body_only=1&module='+module+'&subpanel='+subpanel+'&action=SubPanelViewer&inline=1&record='+id + '&layout_def_key='+submit_data.join('&');
-
+    console.log(url);
```
Comment (mattlorimer) [inline include/SubPanel/SubPanel.js]: ommt logging

File: include/SubPanel/SubPanel.php [hunk 3]
```diff
@@ -400,9 +400,15 @@ function buildSearchQuery()
 	{
 		require_once('include/SubPanel/SubPanelSearchForm.php');
 
-		$module = 'Meetings';
+		if($this->subpanel_id == "history"){
+			$module = 'Meetings';
+			$seed = new Meeting();
+		}else{
+			$module = 'Contacts';
```
Comment (mattlorimer) [inline include/SubPanel/SubPanel.php]: hard coded module need to be changed, to something more dynamic

File: include/SubPanel/SubPanelDefinitions.php [hunk 4]
```diff
@@ -95,8 +95,13 @@ function __construct ($name , $instance_properties , $parent_bean , $reload = fa
 			}
 		}
 		if (!$this->isCollection()){
-			$table = strtolower($instance_properties['module']);
-			$search_query = str_replace('meetings',$table,$search_query);
+			if($parent_bean->module_name == "FP_events"){
```
Comment (mattlorimer) [inline include/SubPanel/SubPanelDefinitions.php]: hard coded module need to be changed, to something more dynamic

File: modules/Accounts/metadata/subpaneldefs.php [hunk 5]
```diff
@@ -125,6 +125,8 @@
                     'get_distinct_data' => true,
                 ),
             ),
+            
+            // suspicious thing
```
Comment (mattlorimer) [inline modules/Accounts/metadata/subpaneldefs.php]: ?

File: themes/SuiteR/css/style.css [hunk 6]
```diff
@@ -5173,6 +5173,9 @@ div.target-list-table {
     padding: 10px;
 }
 
+table tr#delegates_search {
```
Comment (mattlorimer) [inline themes/SuiteR/css/style.css]: is this only relevant to the current theme or any theme?

Comment (shogunpol) [inline themes/SuiteR/css/style.css]: All sorted out :)

________________________________

File: include/EditView/Panels.js [hunk 7]
```diff
@@ -1,6 +1,9 @@
 /*********************************************************************************
```
Comment (mattlorimer) [inline include/EditView/Panels.js]: this is the old licence header

File: config_override.php [hunk 8]
```diff
@@ -1,3 +1,23 @@
 <?php
 /***CONFIGURATOR***/
+$sugar_config['theme_settings']['Suite7']['menu'] = '121212';
```
Comment (mattlorimer) [inline config_override.php]: do not commit this

**Comment by mattlorimer** [comment]
there appears to be lots of file committed that shouldn't be, and most commit messages are the same.

File: include/EditView/Panels.js (hunk 7 above)
Comment (shogunpol) [inline include/EditView/Panels.js]: Dear Matt


This is changed, i will pay attention on this from now.


Jerzy

________________________________

File: config_override.php (hunk 8 above)
Comment (shogunpol) [inline config_override.php]: Dear Matt


I start do not do it, Will and Ashley informed me,

how to improve my commits.



Jerzy


________________________________

Comment (shogunpol) [inline config_override.php]: Dear Matt


This is changed, i was informed by Will today

do not commit files what i do not did changes on myself,

especially:

config.php

config_override.php

.ext.php ones

and all /cache


Jerzy


________________________________

File: modules/FP_events/metadata/subpaneldefs.php [hunk 9]
```diff
@@ -54,7 +54,7 @@
 			'module'=>'Delegates',
 
 			'top_buttons' => array(
-				array('widget_class' => 'SubPanelTopButton_c'),
+				//array('widget_class' => 'SubPanelTopButton_c'),
```
Comment (samus-aran) [inline modules/FP_events/metadata/subpaneldefs.php]: Remove rather than comment

File: include/EditView/Panels.js [hunk 10]
```diff
@@ -10,7 +10,7 @@
  * OF NON INFRINGEMENT OF THIRD PARTY RIGHTS.
  *
  * This program is distributed in the hope that it will be useful, but WITHOUT
- * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * ANY WARRANTY; without even the mplied warranty of MERCHANTABILITY or FITNESS
```
Comment (samus-aran) [inline include/EditView/Panels.js]: typo?

File: modules/FP_events/metadata/subpaneldefs.php (hunk 9 above)
Comment (shogunpol) [inline modules/FP_events/metadata/subpaneldefs.php]: Dear Ashley

Thanks for your comment, i will try next time to remove all old or commented code.

Jerzy
________________________________

Comment (samus-aran) [inline modules/FP_events/metadata/subpaneldefs.php]: Hi Jerzy - please update this to pass this PR review.

File: include/EditView/Panels.js (hunk 10 above)
Comment (samus-aran) [inline include/EditView/Panels.js]: Hi Jerzy - https://suitecrm.com/wiki/index.php/Coding_Standards#License

No, this typo was added by the commit - there is no typo on the Coding Standards.  Update this.

Comment (shogunpol) [inline include/EditView/Panels.js]: Dear Ashley


I'm so sorry for this,

I will fix it as soon as I will start work tomorrow


Jerzy

________________________________

Comment (shogunpol) [inline include/EditView/Panels.js]: Dear Ashley


Changes done, again apologise for delay


Jerzy

________________________________

File: data/SugarBean.php [hunk 11]
```diff
@@ -770,6 +770,9 @@ public static function get_union_related_list($parentbean, $order_by = "", $sort
 
     protected static function build_sub_queries_for_union($subpanel_list, $subpanel_def, $parentbean, $order_by)
     {
+
+        //$subpanel_def->search_query = "(contacts.first_name like '11111111111%' ) AND ( contacts.last_name like '22222222222%')";
+
```
Comment (samus-aran) [inline data/SugarBean.php]: Comments still here - remove this.

File: modules/FP_events/metadata/editviewdefs.php [hunk 12]
```diff
@@ -89,49 +89,49 @@
...
+                                        1 =>
+                                            array (
+                                                'name' => 'decline_redirect',
+                                                'label' => 'LBL_DECLINE_REDIRECT',
+                                            ),
+                                    ),
+                            ),
+                    ),
+            ),
+    );
```
Comment (samus-aran) [inline modules/FP_events/metadata/editviewdefs.php]: Changes like these don't bother unless necessary.

File: modules/FP_events/metadata/listviewdefs.php [hunk 13]
```diff
@@ -1,52 +1,52 @@
...
-    'width' => '9%',
-    'label' => 'LBL_ASSIGNED_TO_NAME',
-    'module' => 'Employees',
-    'id' => 'ASSIGNED_USER_ID',
-    'default' => true,
-  ),
-);
-?>
+$listViewDefs [$module_name] =
+    array (
```
Comment (samus-aran) [inline modules/FP_events/metadata/listviewdefs.php]: Changes like these don't bother unless necessary.

**Comment by samus-aran** [comment]
There are a lot of files changes over the course of this PR - going forward restrict what you push up and remember to always be aware of what files you want up rather than committing up everything you have touched.

**Comment by shogunpol** [comment]
Dear Ashley


The first issue (comment) was removed;

The Second and Third according to your suggestion i leaved untouched;

The last One - i trying to improve my pushing control to Git - obviously still

a lot to improve.


Jerzy

________________________________
